### PR TITLE
fix QVector<bool> packing when number of joints isn't a multiple of 8

### PR DESCRIPTION
**Pull request by sethalves**
- fix a network protocol bug related to model-entity joints

### CODE REVIEW

**Comment by sethalves** [comment]
to test:
- run http://hifi-pet.herokuapp.com/scripts/summonDragonJointCrashAnimated.js and see that the entity-server doesn't crash

**Comment by sethalves** [comment]
https://github.com/highfidelity/hifi/issues/8981

**Comment by sethalves** [comment]
https://highfidelity.fogbugz.com/f/cases/2176/Entities-setAbsoluteJointRotationInObjectFrame-crashes-entity-server

**Comment by hifi-gustavo** [comment]
Build console output for commit: 64cb5d1b1a0d5b0b41b44e99326a1507b685bab5 available at: https://gist.github.com/c7103fd6c42713df39a90a351667c0ad<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8991-64cb5d1.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8991-64cb5d1.dmg

**Comment by Nex-Pro** [comment]
### PR #8991 


**Testing Architecture:** Desktop PC
**CPU:** 4.4 ghz Intel i7
**RAM:** 16 GB
**GPU:** Nvidia gtx 1070 8gb

**Testing OS:** win 10 pro

### FAIL

First time running script is fine. Reloading causes near instant crash of client.
Returning to location of crash invokes further crash. Reset and clear cache did not work.
Ran this numerous times.

**Location of tests:**
sandbox/10000,10000,10000

Multiple splats returned.

**Comment by sethalves** [comment]
QA using the server from this PR, please

**Comment by sethalves** [comment]
I've made a change that will keep interface from this PR from crashing when talking to a server from before this PR, but it will still complain about networking in its log.  Regardless, please re-QA with the server from this PR.

**Comment by hifi-gustavo** [comment]
Build console output for commit: cfea3cba9b0ba9654b871f712676d6595121cf7d available at: https://gist.github.com/c7103fd6c42713df39a90a351667c0ad<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8991-cfea3cb.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8991-cfea3cb.dmg

**Comment by Nex-Pro** [comment]
Testing new build.

File: libraries/octree/src/OctreePacketData.cpp [hunk 1]
```diff
@@ -715,6 +730,10 @@ int OctreePacketData::unpackDataFromBytes(const unsigned char* dataBytes, QVecto
     uint16_t length;
     memcpy(&length, dataBytes, sizeof(uint16_t));
     dataBytes += sizeof(length);
+    if (length * sizeof(bool) > MAX_OCTREE_UNCOMRESSED_PACKET_SIZE) {
```
Comment (ZappoMan) [inline libraries/octree/src/OctreePacketData.cpp]: hmmm - is it really sizeof(bool)? isn't it more like 1 bit per packed bool?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1d0c3d71688d8aa897a8ec93581b815595964186 available at: https://gist.github.com/c7103fd6c42713df39a90a351667c0ad<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8991-1d0c3d7.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8991-1d0c3d7.dmg

**Comment by Nex-Pro** [comment]
#### SUCCESS

Tried this with Michelle server + client and also with just the server and using current release. 

No crashes.

**Pull request merged**